jmix-flowui/flowui/src/main/java/io/jmix/flowui/facet/urlqueryparameters/GenericFilterUrlQueryParametersBinder.java

```diff
@@ -262,6 +263,10 @@ protected boolean isPermitted(DataLoader dataLoader, FilterComponent filterCompo
         return true;
     }
 
+    private MetadataTools recieveMetadataTools() {
```

Comment:
1. We **do not** use private modifier
2. Use the same naming as for other similar methods, either `getFoo()` or `foo()`
3. There is no point in this method right now, because it calls `applicationContext.getBean(MetadataTools.class)` every time. Take a look at `singleFilterSupport` in this class and do the same - lazy initialization.
4. Place your method near similar `getSingleFilterSupport()` since they are logically the same.

Reply:
fixed, but it is strange practice to use only protected modifiers. It obviously conflicts with Open-Close principle of SOLID principes.